makefile

```diff
@@ -185,6 +185,7 @@ build/src/libultra_boot_O2/%.o: CC := $(CC_OLD)
 
 build/src/boot/%.o: CC := python3 tools/asm_processor/build.py $(CC) -- $(AS) $(ASFLAGS) --
 build/src/code/%.o: CC := python3 tools/asm_processor/build.py $(CC) -- $(AS) $(ASFLAGS) --
+build/src/libultra_code/%.o: CC := python3 tools/asm_processor/build.py $(CC_OLD) -- $(AS) $(ASFLAGS) --
```

Comment:
Since this uses CC_OLD directly, the line above setting CC to CC_OLD can be removed.

Reply:
Gonna have to disagree on this one--the line above *should* set CC to CC_OLD for this directory, with this line just saying CC, but it actually doesn't because make is weird. So, changing CC to CC_OLD on this line is needed to make it work, but the earlier group is still defining what directories need to be compiled with CC_OLD. Ideally this would all be defined in a more organized fashion, with separate lists for which directories need asm_processor, which need CC_OLD, etc., where they can all stack properly.

Reply:
This line redefines CC so the previous one should have literally no effect, unless I'm missing something.

Reply:
You're absolutely right that it's redundant. The thing is if we remove the first copy, then it won't be with all the other files which need CC_OLD. It'll have been changed to CC_OLD somewhere else where people aren't expecting. If the build toolchain gets changed in the future to a more sane approach, the user may not notice that this directory is CC_OLD.

Also there's the matter that it should work like this:
CC = CC_OLD
CC = asm_processor CC
This should result in CC = asm_processor CC_OLD. But due to make being weird, it doesn't. So the second one is only CC_OLD to workaround make, not because we actually want it to be defined as CC_OLD there.

This is not the hill I want to die on, so if you insist I'll change it.

Reply:
I've commented it out and left a comment explaining.